src/pipe_log.rs

```diff
@@ -120,4 +191,11 @@ pub trait PipeLog: Sized {
     ///
     /// Returns the number of deleted files.
     fn purge_to(&self, file_id: FileId) -> Result<usize>;
+
+    /// Returns the `[Version]` corresponding to the given `[FileId]`
+    fn fetch_format_version(&self, file_id: FileId) -> Result<Version>;
+
+    /// Returns `[LogFileContext]` of the active file in the specific
+    /// log queue.
+    fn fetch_active_file(&self, queue: LogQueue) -> Result<LogFileContext>;
```

Comment:
Don't return `Result`, the same way as `file_span`, we can assume there's at least one file in collection.